backend/netstorage/netstorage.go

```diff
@@ -0,0 +1,1369 @@
+// Package netstorage provides an interface to Akamai NetStorage API
+package netstorage
+
+import (
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/xml"
+	"io"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"runtime/debug"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/rclone/rclone/fs"
+	"github.com/rclone/rclone/fs/config/configmap"
+	"github.com/rclone/rclone/fs/config/configstruct"
+	"github.com/rclone/rclone/fs/config/obscure"
+	"github.com/rclone/rclone/fs/fshttp"
+	"github.com/rclone/rclone/fs/hash"
+	"github.com/rclone/rclone/fs/walk"
+	"github.com/rclone/rclone/lib/pacer"
+	"github.com/rclone/rclone/lib/rest"
+	"golang.org/x/net/html/charset"
+)
+
+// Constants
+const (
+	timeFormatIn  = time.UnixDate
+	minSleep      = 10 * time.Millisecond
+	maxSleep      = 2 * time.Second
+	decayConstant = 2 // bigger for slower decay, exponential
+)
+
+func init() {
+	fsi := &fs.RegInfo{
+		Name:        "netstorage",
+		Description: "Akamai NetStorage",
+		NewFs:       NewFs,
+		CommandHelp: commandHelp,
+		Options: []fs.Option{{
+			Name: "protocol",
+			Help: "Select between HTTP or HTTPS protocol. Most users should choose HTTPS, which is the default. HTTP is provided primarily for debugging purposes.",
+			Examples: []fs.OptionExample{{
+				Value: "http",
+				Help:  "HTTP protocol",
+			}, {
+				Value: "https",
+				Help:  "HTTPS protocol",
+			}},
+			Required: true,
```

Comment:
Do you want to set a `Default: "https"` here and remove the required? That might make more sense.

This should probably have `Advanced: true` also as it isn't needed for normal operations.